### DISCUSSION TITLE
基于众包的视频赞助商广告标注及跳过功能

### DISCUSSION THREAD

**Discussion by willia0** [the1812/Bilibili-Evolved#4507] [功能建议 / Ideas]
截止2023年10月，b站的每九个视频中就有十个护颈枕和电动牙刷的赞助广告，游戏区视频中加速器广告甚至能占到视频时长的一半。不感兴趣的用户只能根据弹幕里的“空降坐标”或是进度条里的“高能时刻”大致判断广告的结束时间并手动跳过，令人十分头疼。Youtube上能够实现自动跳过广告功能的[SponsorBlock](https://github.com/ajayyy/SponsorBlock#sponsorblock)插件也暂无增加支持bilibili的计划。[Bilibili-Evolved](https://github.com/the1812/Bilibili-Evolved)作为最受欢迎的bilibili插件之一，应该有足够的用户基数来维护一个众包的广告标记数据集。虽然听起来工作量很大，但是否有可能通过组件/插件等方式实现这一功能呢？

**Comment by the1812** [comment]
咋众包, Bilibili-Evolved 没有独立的后端也没有数据库, 不能指望人人都会 Pull Request 吧

**Reply by WakelessSloth56** [reply]
可以考虑用 Serverless（e.g. Cloudflare Workers）来实现
